lib/src/message.dart

```diff
@@ -138,13 +138,16 @@ abstract class Message {
 Map<String, String> _adjustHeaders(
     Map<String, String> headers, Encoding encoding) {
   if (headers == null) headers = const {};
-  if (encoding == null) return headers;
-  if (headers['content-type'] == null) {
-    return addHeader(headers, 'content-type',
+
+  var caseInsensitiveHeaders = new CaseInsensitiveMap.from(headers);
```

Comment:
Don't copy the headers map if `encoding` is `null`.

Also, if you just re-assign to `headers`, you can avoid wrapping the const map in the case that `headers` was null.